gadget/position_test.go

```diff
@@ -137,40 +136,28 @@ volumes:
 	c.Assert(err, IsNil)
 
 	c.Assert(v, DeepEquals, &gadget.PositionedVolume{
-		Size: 1101 * gadget.SizeMiB,
-		Structures: []gadget.PositionedStructure{
+		Volume: vol,
+		Size:   1101 * gadget.SizeMiB,
+		PositionedStructure: []gadget.PositionedStructure{
 			{
-				VolumeStructure: &gadget.VolumeStructure{
-					Size: 400 * gadget.SizeMiB,
-					Type: "00000000-0000-0000-0000-dd00deadbeef",
-				},
-				StartOffset: 1 * gadget.SizeMiB,
-				Index:       0,
+				VolumeStructure: &vol.Structure[0],
+				StartOffset:     1 * gadget.SizeMiB,
+				Index:           0,
 			},
 			{
-				VolumeStructure: &gadget.VolumeStructure{
-					Size: 500 * gadget.SizeMiB,
-					Type: "00000000-0000-0000-0000-cc00deadbeef",
-					Role: "system-data",
-				},
-				StartOffset: 401 * gadget.SizeMiB,
-				Index:       1,
+				VolumeStructure: &vol.Structure[1],
+				StartOffset:     401 * gadget.SizeMiB,
+				Index:           1,
 			},
 			{
-				VolumeStructure: &gadget.VolumeStructure{
-					Size: 100 * gadget.SizeMiB,
-					Type: "00000000-0000-0000-0000-bb00deadbeef",
-				},
-				StartOffset: 901 * gadget.SizeMiB,
-				Index:       2,
+				VolumeStructure: &vol.Structure[2],
+				StartOffset:     901 * gadget.SizeMiB,
+				Index:           2,
 			},
 			{
-				VolumeStructure: &gadget.VolumeStructure{
-					Size: 100 * gadget.SizeMiB,
-					Type: "00000000-0000-0000-0000-aa00deadbeef",
-				},
-				StartOffset: 1001 * gadget.SizeMiB,
-				Index:       3,
+				VolumeStructure: &vol.Structure[3],
```

Comment:
Would it make sense to assert on len(vol.Structure) == 4 after `vol := mustParseVolume(...` (above and in other tests)?

Reply:
Pushed